Adopt updated interfaces

mwyrzykowski opened the pull request:
#### 8580e5313691c33f0465a30fb23a9be89bd663b3
<pre>
Adopt updated interfaces
<a href="https://bugs.webkit.org/show_bug.cgi?id=301971">https://bugs.webkit.org/show_bug.cgi?id=301971</a>
<a href="https://rdar.apple.com/164047413">rdar://164047413</a>

Reviewed by Etienne Segonzac.

Migrate from earlier prototype frameworks to updated prototype frameworks.

The updated prototype frameworks support meshes, texturing, and materials,
but no animations yet.

Texture upload support is performed primitively now, upcoming
PR will adopt RE texture importer. That is why placeholder
code only supports 8 bits per channel.

* Source/WebCore/Modules/Model/Implementation/DDMeshImpl.cpp:
(WebCore::DDModel::convertToBacking):
(WebCore::DDModel::convert):
(WebCore::DDModel::DDMeshImpl::updateTexture):
(WebCore::DDModel::DDMeshImpl::addMesh): Deleted.
(WebCore::DDModel::DDMeshImpl::addTexture): Deleted.
* Source/WebCore/Modules/Model/Implementation/DDMeshImpl.h:
* Source/WebCore/Modules/Model/Implementation/DDMeshImpl.mm: Added.
(WebCore::DDModel::DDMeshImpl::DDMeshImpl):
(WebCore::DDModel::DDMeshImpl::setLabelInternal):
(WebCore::DDModel::convert):
(WebCore::DDModel::DDMeshImpl::update):
(WebCore::DDModel::DDMeshImpl::updateTexture):
(WebCore::DDModel::convertArray):
(WebCore::DDModel::DDMeshImpl::updateMaterial):
(WebCore::DDModel::DDMeshImpl::render):
(WebCore::DDModel::DDMeshImpl::setEntityTransform):
(WebCore::DDModel::DDMeshImpl::entityTransform const):
(WebCore::DDModel::DDMeshImpl::setCameraDistance):
(WebCore::DDModel::DDMeshImpl::play):
(WebCore::DDModel::DDMeshImpl::ioSurfaceHandles):
* Source/WebCore/Modules/Model/InternalAPI/DDBridgeMeshDescriptor.h: Renamed from Source/WebCore/Modules/Model/InternalAPI/DDReplaceVertices.h.
* Source/WebCore/Modules/Model/InternalAPI/DDMesh.h:
* Source/WebCore/Modules/Model/InternalAPI/DDMeshDescriptor.h:
(): Deleted.
* Source/WebCore/Modules/Model/InternalAPI/DDModel.serialization.in:
* Source/WebCore/Modules/Model/InternalAPI/DDUpdateMeshDescriptor.h:
* Source/WebCore/Modules/Model/InternalAPI/DDVertexAttributeFormat.h:
* Source/WebCore/Modules/Model/InternalAPI/DDVertexLayout.h:
* Source/WebCore/Modules/Model/ModelDDInlineConverters.h:
(WebCore::toCpp):
(WebCore::toCppVertexAttributes): Deleted.
(WebCore::toCppVertexLayouts): Deleted.
(WebCore::toVector): Deleted.
* Source/WebCore/Modules/Model/ModelDDTypes.h:
* Source/WebCore/Modules/contact-picker/ContactsManager.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
* Source/WebCore/Modules/model-element/DDModelPlayer.h:
* Source/WebCore/Modules/model-element/DDModelPlayer.mm:
(WebCore::DDModelPlayer::load):
* Source/WebCore/Scripts/generate-unified-sources.sh:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/UnifiedSources-output.xcfilelist:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/UsdModelLoader.swift: Removed.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/DDMesh.h:
* Source/WebGPU/WebGPU/DDMesh.mm:
(WebGPU::DDMesh::DDMesh):
(WebGPU::DDMesh::update):
(WebGPU::DDMesh::updateTexture):
(WebGPU::DDMesh::updateMaterial):
(wgpuDDMeshUpdate):
(wgpuDDTextureUpdate):
(wgpuDDMaterialUpdate):
(): Deleted.
(WebGPU::convertDescriptor): Deleted.
(WebGPU::convertFloats): Deleted.
(WebGPU::convertPart): Deleted.
(WebGPU::convertUint8s): Deleted.
(WebGPU::convert): Deleted.
(WebGPU::DDMesh::addMesh): Deleted.
(WebGPU::DDMesh::addTexture): Deleted.
(WebGPU::DDMesh::addMaterial): Deleted.
(wgpuDDMeshAdd): Deleted.
(wgpuDDTextureAdd): Deleted.
(wgpuDDMaterialAdd): Deleted.
* Source/WebGPU/WebGPU/DDModelTypes.h:
* Source/WebGPU/WebGPU/USDModel.swift: Renamed from Source/WebGPU/WebGPU/UsdModelRenderer.swift.
(DDBridgeChainedFloat4x4.transform):
(DDBridgeChainedFloat4x4.next):
(DDBridgeUpdateMesh.instanceTransforms):
(NodeStore.nodeDefinitionStore):
(mapSemantic(_:)):
(_Proto_LowLevelMeshResource_v1.fromLlmDescriptor(_:)):
(Helper.isNonZero(_:)):
(Helper.makeDirectTextureDescriptorFromMTLTexture(_:)):
(Helper.pixelFormat):
(sgDataTypeFromInputOutputDataType(_:)):
(nodes):
(allNodes):
(connections):
(customSurfaceShaderParameterBindings):
(assetPathToPropertyName):
(pbrInputNameToType(_:)):
(_Proto_LowLevelMeshResource_v1.replaceVertexData(_:)):
(_Proto_LowLevelMeshResource_v1.replaceIndexData(_:)):
(_Proto_LowLevelMeshResource_v1.replaceData(_:replaceIndexData:replaceVertexData:)):
(DDBridgeReceiver.meshInstances):
(DDBridgeReceiver.meshResources):
(DDBridgeReceiver.camera):
(DDBridgeReceiver.renderer):
(DDBridgeReceiver.materials):
(DDBridgeReceiver.textures):
(DDBridgeReceiver.textureFromPath):
(DDBridgeReceiver.modelTransform):
(DDBridgeReceiver.modelDistance):
(DDBridgeReceiver.initRenderer(_:)):
(DDBridgeReceiver.render(with:)):
(DDBridgeReceiver.updateTextureAsync(_:)):
(DDBridgeReceiver.updateTexture(_:)):
(DDBridgeReceiver.updateMaterialAsync(_:)):
(DDBridgeReceiver.updateMaterial(_:)):
(DDBridgeReceiver.updateMeshAsync(_:)):
(DDBridgeReceiver.updateMesh(_:)):
(DDBridgeReceiver.setTransformAsync(_:)):
(DDBridgeReceiver.setTransform(_:)):
(DDBridgeReceiver.setCameraDistance(_:)):
(DDBridgeReceiver.setPlaying(_:)):
(Converter.convertSemantic(_:)):
(Converter.webAttributesFromAttributes(_:)):
(Converter.webLayoutsFromLayouts(_:)):
(Converter.webMeshDescriptorFromMeshDescriptor(_:)):
(Converter.webPartsFromParts(_:)):
(Converter.convert(_:)):
(Converter.webRequestInstanceTransforms):
(Converter.descriptor):
(USDModelLoader.meshUpdated(_:)):
(USDModelLoader.meshDestroyed(_:)):
(USDModelLoader.materialUpdated(_:)):
(USDModelLoader.materialDestroyed(_:)):
(USDModelLoader.textureUpdated(_:)):
(USDModelLoader.textureDestroyed(_:)):
(USDModelLoader.loadModel(from:)):
(USDModelLoader.update(_:)):
(DDBridgeModelLoader.loader):
(DDBridgeModelLoader.modelUpdated):
(DDBridgeModelLoader.textureUpdatedCallback):
(DDBridgeModelLoader.materialUpdatedCallback):
(DDBridgeModelLoader.retainedRequests):
(DDBridgeModelLoader.loadModel(from:)):
(DDBridgeModelLoader.update(_:)):
(DDBridgeModelLoader.requestCompleted(_:)):
(DDBridgeModelLoader.updateMesh(_:)):
(DDBridgeModelLoader.updateTexture(_:)):
(DDBridgeModelLoader.updateMaterial(_:)):
* Source/WebGPU/WebGPU/WebGPUExt.h:
(): Deleted.
* Source/WebKit/GPUProcess/graphics/Model/RemoteDDMesh.cpp:
(WebKit::RemoteDDMesh::addMesh): Deleted.
(WebKit::RemoteDDMesh::addTexture): Deleted.
(WebKit::RemoteDDMesh::addMaterial): Deleted.
* Source/WebKit/GPUProcess/graphics/Model/RemoteDDMesh.h:
* Source/WebKit/GPUProcess/graphics/Model/RemoteDDMesh.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/WebProcess/GPU/graphics/Model/RemoteDDMeshProxy.cpp:
(WebKit::DDModel::computeMinAndMaxCorners):
(WebKit::DDModel::RemoteDDMeshProxy::update):
(WebKit::DDModel::RemoteDDMeshProxy::addMesh): Deleted.
(WebKit::DDModel::RemoteDDMeshProxy::addTexture): Deleted.
(WebKit::DDModel::RemoteDDMeshProxy::addMaterial): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/Model/RemoteDDMeshProxy.h:

* Source/WebKit/Scripts/webkit/opaque_ipc_types.tracking.in:
Rename of types

Canonical link: <a href="https://commits.webkit.org/303656@main">https://commits.webkit.org/303656@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5fe2b1fa416bb3eec7b20047e6bbe87fe17cd0d4

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133081 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/5582 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44186 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/140625 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/85118 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/1e5a3d0b-076c-4cbf-83d3-d6370e75838b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/134951 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6074 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5445 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/101775 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/69145 "Passed tests") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/26dbfbae-d368-4f2f-8bb0-ed5b95aac731) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136028 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/4281 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/119265 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82573 "Passed tests") | | ⏳ 🛠 vision-apple 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/132430 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/4166 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/1754 "Passed tests") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/125153 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/113239 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/37381 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143272 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/131590 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5252 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/37960 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/110155 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5334 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4511 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110335 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27981 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4051 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115522 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/58875 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5307 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/33874 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/164557 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5148 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/68759 "Built successfully") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/25/builds/42886 "Build was cancelled. Recent messages:") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/5396 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5263 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebGPU/WebGPU/USDModel.swift:
```diff
@@ -1251,88 +1219,87 @@ extension DDBridgeReceiver {
+        }
+
+        for instance in scene.meshInstances {
+            let (_, originalTransform) = instancesAndTransforms.first(where: { $0.0 === instance })!
```
@rr-codes style checker is stating:
```
Source/WebGPU/WebGPU/USDModel.swift:1256: error: [NeverForceUnwrap] do not force unwrap 'instancesAndTransforms.first(where: { $0.0 === instance })'
```
but I don't see where the force unwrap occurs?

rr-codes replied:
it’s the exclamation mark at the end of 

```
let (_, originalTransform) = instancesAndTransforms.first(where: { $0.0 === instance })!
```

rr-codes replied:
(since `first` returns an optional)

mwyrzykowski replied:
Oh that makes sense :)

mwyrzykowski commented on Source/WebGPU/WebGPU/WebGPUExt.h:
```diff
@@ -68,224 +68,12 @@ typedef enum WGPUSTypeExtended {
     WGPUSTypeExtended_Force32 = 0x7FFFFFFF
 } WGPUSTypeExtended;
 
-typedef struct WGPUDDMeshPart {
```
instead of these redundant conversions, I just pass Objective-C / Swift types across the WebCore -> WebGPU framework boundary

etiennesegonzac commented on Source/WebCore/Modules/Model/Implementation/DDMeshImpl.mm:
```diff
@@ -0,0 +1,239 @@
+
+    for (uint32_t i = 1; i < count; ++i) {
+        DDBridgeChainedFloat4x4 *next = [[DDBridgeChainedFloat4x4 alloc] initWithTransform:v[i]];
+        current.next = next;
```
I know it's not new code but... shouldn't we be updating `current`?

mwyrzykowski replied:
Oh yes, that's not going to work for any more than 2 instances, thank you for catching

etiennesegonzac commented on Source/WebGPU/WebGPU/DDMesh.mm:
```diff
@@ -95,8 +64,12 @@
-#if ENABLE(WEBGPU_SWIFT)
-    m_ddReceiver = [[DDBridgeReceiver alloc] initWithDevice:instance.device()];
+#if ENABLE(GPU_PROCESS_MODEL)
+    DDBridgeReceiver* ddReceiver = [[DDBridgeReceiver alloc] initWithDevice:instance.device()];
```
RetainPtr?

mwyrzykowski replied:
`RetainPtr` is not needed since WebGPU.framework is ARC by default

mwyrzykowski replied:
Oh this file is in WebCore unlike the others, but I made it ARC

mwyrzykowski replied:
https://github.com/WebKit/WebKit/pull/53644/files#r2561274166

etiennesegonzac commented on Source/WebGPU/WebGPU/DDMesh.h:
```diff
@@ -78,7 +78,7 @@ class DDMesh : public ThreadSafeRefCounted<DDMesh>, public WGPUDDMeshImpl {
 
-#if ENABLE(WEBGPU_SWIFT)
+#if ENABLE(GPU_PROCESS_MODEL)
     DDBridgeReceiver* m_ddReceiver;
```
RetainPtr?

mwyrzykowski replied:
`RetainPtr` is not needed since WebGPU.framework is ARC by default

etiennesegonzac commented on Source/WebGPU/WebGPU/DDMesh.h:
```diff
@@ -78,7 +78,7 @@ class DDMesh : public ThreadSafeRefCounted<DDMesh>, public WGPUDDMeshImpl {
     WGPUDDCreateMeshDescriptor m_descriptor;
     NSMutableArray<id<MTLTexture>>* m_textures { nil };
```
same

etiennesegonzac reviewed (approved):
Let's add a few lines to the commit message to explain the move between frameworks.
I understand there are ongoing discussions on this topic, but I agree we shouldn't block the framework adoption on this.

mwyrzykowski commented on Source/WebCore/SourcesCocoa.txt:
```diff
@@ -81,6 +81,7 @@ JSSpeechSynthesisErrorCode.cpp
 JSSpeechSynthesisErrorEvent.cpp
 JSSpeechSynthesisErrorEventInit.cpp
 JSSpeechSynthesisEventInit.cpp
+Modules/Model/Implementation/DDMeshImpl.mm
```
the .mm file is ARC

mwyrzykowski commented on Source/WebKit/Scripts/webkit/opaque_ipc_types.tracking.in:
```diff
@@ -233,8 +233,6 @@
     [Legacy] MessageParam RemoteQueue.WriteTextureWithCopy data
     [Legacy] MessageParam WebPageProxy.BindRemoteAccessibilityFrames dataToken
     [Legacy] MessageParam WebSWServerConnection.SubscribeToPushService applicationServerKey
-    [Legacy] StructureParam WebKit::DDModel::DDReplaceVertices.buffer
```
Net non-whitespace lines changed here is zero

mwyrzykowski committed:
Adopt updated interfaces
https://bugs.webkit.org/show_bug.cgi?id=301971
rdar://164047413

Reviewed by Etienne Segonzac.

Migrate from earlier prototype frameworks to updated prototype frameworks.

The updated prototype frameworks support meshes, texturing, and materials,
but no animations yet.

Texture upload support is performed primitively now, upcoming
PR will adopt RE texture importer. That is why placeholder
code only supports 8 bits per channel.

* Source/WebCore/Modules/Model/Implementation/DDMeshImpl.cpp:
(WebCore::DDModel::convertToBacking):
(WebCore::DDModel::convert):
(WebCore::DDModel::DDMeshImpl::updateTexture):
(WebCore::DDModel::DDMeshImpl::addMesh): Deleted.
(WebCore::DDModel::DDMeshImpl::addTexture): Deleted.
* Source/WebCore/Modules/Model/Implementation/DDMeshImpl.h:
* Source/WebCore/Modules/Model/Implementation/DDMeshImpl.mm: Added.
(WebCore::DDModel::DDMeshImpl::DDMeshImpl):
(WebCore::DDModel::DDMeshImpl::setLabelInternal):
(WebCore::DDModel::convert):
(WebCore::DDModel::DDMeshImpl::update):
(WebCore::DDModel::DDMeshImpl::updateTexture):
(WebCore::DDModel::convertArray):
(WebCore::DDModel::DDMeshImpl::updateMaterial):
(WebCore::DDModel::DDMeshImpl::render):
(WebCore::DDModel::DDMeshImpl::setEntityTransform):
(WebCore::DDModel::DDMeshImpl::entityTransform const):
(WebCore::DDModel::DDMeshImpl::setCameraDistance):
(WebCore::DDModel::DDMeshImpl::play):
(WebCore::DDModel::DDMeshImpl::ioSurfaceHandles):
* Source/WebCore/Modules/Model/InternalAPI/DDBridgeMeshDescriptor.h: Renamed from Source/WebCore/Modules/Model/InternalAPI/DDReplaceVertices.h.
* Source/WebCore/Modules/Model/InternalAPI/DDMesh.h:
* Source/WebCore/Modules/Model/InternalAPI/DDMeshDescriptor.h:
(): Deleted.
* Source/WebCore/Modules/Model/InternalAPI/DDModel.serialization.in:
* Source/WebCore/Modules/Model/InternalAPI/DDUpdateMeshDescriptor.h:
* Source/WebCore/Modules/Model/InternalAPI/DDVertexAttributeFormat.h:
* Source/WebCore/Modules/Model/InternalAPI/DDVertexLayout.h:
* Source/WebCore/Modules/Model/ModelDDInlineConverters.h:
(WebCore::toCpp):
(WebCore::toCppVertexAttributes): Deleted.
(WebCore::toCppVertexLayouts): Deleted.
(WebCore::toVector): Deleted.
* Source/WebCore/Modules/Model/ModelDDTypes.h:
* Source/WebCore/Modules/contact-picker/ContactsManager.cpp:
* Source/WebCore/Modules/mediastream/MediaStreamTrack.cpp:
* Source/WebCore/Modules/model-element/DDModelPlayer.h:
* Source/WebCore/Modules/model-element/DDModelPlayer.mm:
(WebCore::DDModelPlayer::load):
* Source/WebCore/Scripts/generate-unified-sources.sh:
* Source/WebCore/Sources.txt:
* Source/WebCore/SourcesCocoa.txt:
* Source/WebCore/UnifiedSources-output.xcfilelist:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/html/canvas/UsdModelLoader.swift: Removed.
* Source/WebGPU/WebGPU.xcodeproj/project.pbxproj:
* Source/WebGPU/WebGPU/DDMesh.h:
* Source/WebGPU/WebGPU/DDMesh.mm:
(WebGPU::DDMesh::DDMesh):
(WebGPU::DDMesh::update):
(WebGPU::DDMesh::updateTexture):
(WebGPU::DDMesh::updateMaterial):
(wgpuDDMeshUpdate):
(wgpuDDTextureUpdate):
(wgpuDDMaterialUpdate):
(): Deleted.
(WebGPU::convertDescriptor): Deleted.
(WebGPU::convertFloats): Deleted.
(WebGPU::convertPart): Deleted.
(WebGPU::convertUint8s): Deleted.
(WebGPU::convert): Deleted.
(WebGPU::DDMesh::addMesh): Deleted.
(WebGPU::DDMesh::addTexture): Deleted.
(WebGPU::DDMesh::addMaterial): Deleted.
(wgpuDDMeshAdd): Deleted.
(wgpuDDTextureAdd): Deleted.
(wgpuDDMaterialAdd): Deleted.
* Source/WebGPU/WebGPU/DDModelTypes.h:
* Source/WebGPU/WebGPU/USDModel.swift: Renamed from Source/WebGPU/WebGPU/UsdModelRenderer.swift.
(DDBridgeChainedFloat4x4.transform):
(DDBridgeChainedFloat4x4.next):
(DDBridgeUpdateMesh.instanceTransforms):
(NodeStore.nodeDefinitionStore):
(mapSemantic(_:)):
(_Proto_LowLevelMeshResource_v1.fromLlmDescriptor(_:)):
(Helper.isNonZero(_:)):
(Helper.makeDirectTextureDescriptorFromMTLTexture(_:)):
(Helper.pixelFormat):
(sgDataTypeFromInputOutputDataType(_:)):
(nodes):
(allNodes):
(connections):
(customSurfaceShaderParameterBindings):
(assetPathToPropertyName):
(pbrInputNameToType(_:)):
(_Proto_LowLevelMeshResource_v1.replaceVertexData(_:)):
(_Proto_LowLevelMeshResource_v1.replaceIndexData(_:)):
(_Proto_LowLevelMeshResource_v1.replaceData(_:replaceIndexData:replaceVertexData:)):
(DDBridgeReceiver.meshInstances):
(DDBridgeReceiver.meshResources):
(DDBridgeReceiver.camera):
(DDBridgeReceiver.renderer):
(DDBridgeReceiver.materials):
(DDBridgeReceiver.textures):
(DDBridgeReceiver.textureFromPath):
(DDBridgeReceiver.modelTransform):
(DDBridgeReceiver.modelDistance):
(DDBridgeReceiver.initRenderer(_:)):
(DDBridgeReceiver.render(with:)):
(DDBridgeReceiver.updateTextureAsync(_:)):
(DDBridgeReceiver.updateTexture(_:)):
(DDBridgeReceiver.updateMaterialAsync(_:)):
(DDBridgeReceiver.updateMaterial(_:)):
(DDBridgeReceiver.updateMeshAsync(_:)):
(DDBridgeReceiver.updateMesh(_:)):
(DDBridgeReceiver.setTransformAsync(_:)):
(DDBridgeReceiver.setTransform(_:)):
(DDBridgeReceiver.setCameraDistance(_:)):
(DDBridgeReceiver.setPlaying(_:)):
(Converter.convertSemantic(_:)):
(Converter.webAttributesFromAttributes(_:)):
(Converter.webLayoutsFromLayouts(_:)):
(Converter.webMeshDescriptorFromMeshDescriptor(_:)):
(Converter.webPartsFromParts(_:)):
(Converter.convert(_:)):
(Converter.webRequestInstanceTransforms):
(Converter.descriptor):
(USDModelLoader.meshUpdated(_:)):
(USDModelLoader.meshDestroyed(_:)):
(USDModelLoader.materialUpdated(_:)):
(USDModelLoader.materialDestroyed(_:)):
(USDModelLoader.textureUpdated(_:)):
(USDModelLoader.textureDestroyed(_:)):
(USDModelLoader.loadModel(from:)):
(USDModelLoader.update(_:)):
(DDBridgeModelLoader.loader):
(DDBridgeModelLoader.modelUpdated):
(DDBridgeModelLoader.textureUpdatedCallback):
(DDBridgeModelLoader.materialUpdatedCallback):
(DDBridgeModelLoader.retainedRequests):
(DDBridgeModelLoader.loadModel(from:)):
(DDBridgeModelLoader.update(_:)):
(DDBridgeModelLoader.requestCompleted(_:)):
(DDBridgeModelLoader.updateMesh(_:)):
(DDBridgeModelLoader.updateTexture(_:)):
(DDBridgeModelLoader.updateMaterial(_:)):
* Source/WebGPU/WebGPU/WebGPUExt.h:
(): Deleted.
* Source/WebKit/GPUProcess/graphics/Model/RemoteDDMesh.cpp:
(WebKit::RemoteDDMesh::addMesh): Deleted.
(WebKit::RemoteDDMesh::addTexture): Deleted.
(WebKit::RemoteDDMesh::addMaterial): Deleted.
* Source/WebKit/GPUProcess/graphics/Model/RemoteDDMesh.h:
* Source/WebKit/GPUProcess/graphics/Model/RemoteDDMesh.messages.in:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/WebProcess/GPU/graphics/Model/RemoteDDMeshProxy.cpp:
(WebKit::DDModel::computeMinAndMaxCorners):
(WebKit::DDModel::RemoteDDMeshProxy::update):
(WebKit::DDModel::RemoteDDMeshProxy::addMesh): Deleted.
(WebKit::DDModel::RemoteDDMeshProxy::addTexture): Deleted.
(WebKit::DDModel::RemoteDDMeshProxy::addMaterial): Deleted.
* Source/WebKit/WebProcess/GPU/graphics/Model/RemoteDDMeshProxy.h:

* Source/WebKit/Scripts/webkit/opaque_ipc_types.tracking.in:
Rename of types

Canonical link: https://commits.webkit.org/303656@main
